lib/ret_web/channels/hub_channel.ex

```diff
@@ -185,12 +186,23 @@ defmodule RetWeb.HubChannel do
     {:reply, {:ok, %{has_remaining_subscriptions: has_remaining_subscriptions}}, socket}
   end
 
-  def handle_in("sign_in", %{"token" => token}, socket) do
+  def handle_in("sign_in", %{"token" => token} = payload, socket) do
+    creator_assignment_token = payload["creator_assignment_token"]
+
     case Ret.Guardian.resource_from_token(token) do
       {:ok, %Account{} = account, _claims} ->
         socket = Guardian.Phoenix.Socket.put_current_resource(socket, account)
 
-        hub = socket |> hub_for_socket
+        hub = socket |> hub_for_socket |> Repo.preload(@hub_preloads)
```

Comment:
We're preloading a lot of things for just a `sign_in`. Maybe just restrict preloads to `created_by_account` in this case?

Reply:
i think this would be an optimization, and could introduce tricky bugs if we change the dependent objects over time, so I'd suggest we leave it as-is until we determine if any of these loads are bottlenecks. as i mentioned i'm generally not nervous about any of these kinds of queries over the long haul (hash indexes will be fast, we can add caching, we can add read repliacs, etc) many-to-many join tables are the only things i'd be a bit careful about if we expect a lot of entries since those may bottleneck out without sharding the data.